pkg/validations/validations.go

```diff
@@ -15,7 +15,7 @@ import (
 
 const (
 	dnsNameRegex             = "^([a-z0-9]+(-[a-z0-9]+)*[.])+[a-z]{2,}$"
-	hostnameRegex            = `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`
+	hostnameRegex            = `^[a-z0-9][a-z0-9\-.]{0,62}[a-z0-9]$`
```

Comment:
https://sourcegraph.com/github.com/openshift/installer@bdab9b334f23f2ef1f277fced734f209d7840ae6/-/blob/vendor/k8s.io/apimachinery/pkg/util/validation/validation.go?L178

for subdomain

Reply:
Good catch, I've escaped the period. Unsure how strict we want to be about the hostname such as the example you mentioned.